lib/openssl.rb

```diff
@@ -20,3 +20,18 @@
 require 'openssl/x509'
 require 'openssl/ssl'
 require 'openssl/pkcs5'
+
+module OpenSSL
+  # call-seq:
+  #   OpenSSL.secure_compare(string, string) -> boolean
+  #
+  # Constant time memory comparison. Inputs are hashed using SHA-256 to prevent
+  # leaking the length of the secret. Returns +true+ if the strings are
+  # identical, +false+ otherwise.
+  def secure_compare(a, b)
```

Comment:
Is there any reason why we can't just use `def self.secure_compare`? rather than `module_function`.

Reply:
`module_function` also makes it a private instance method, so that if someone does `include OpenSSL`, then they call `secure_compare` (instead of `OpenSSL.secure_compare`).  FWIW, `Rack::Utils.secure_compare` also uses `module_function`.